relay-server/src/services/projects/cache/service.rs

```diff
@@ -0,0 +1,218 @@
+use std::sync::Arc;
+
+use relay_base_schema::project::ProjectKey;
+use relay_config::Config;
+use relay_statsd::metric;
+use relay_system::Service;
+use tokio::sync::{broadcast, mpsc};
+
+use crate::services::projects::cache::handle::ProjectCacheHandle;
+use crate::services::projects::cache::state::{CompletedFetch, Fetch, ProjectStore};
+use crate::services::projects::project::ProjectState;
+use crate::services::projects::source::ProjectSource;
+use crate::statsd::RelayTimers;
+
+/// Size of the broadcast channel for project events.
+///
+/// This is set to a value which theoretically should never be reachable,
+/// the number of events is approximately bounded by the amount of projects
+/// receiving events.
+///
+/// It is set to such a large amount because receivers of events currently
+/// do not deal with lags in the channel gracefuly.
+const PROJECT_EVENTS_CHANNEL_SIZE: usize = 512_000;
+
+/// A cache for projects, which allows concurrent access to the cached projects.
```

Comment:
Is this docstring accurate? I believe `ProjectCache` here is a kind of message?

Reply:
Since the service is usually interacted with `Addr<Message>`, e.g. `Addr<ProjectCache>` the service documentation is on the message.